# Load PlayStation sales data from Kaggle dataset
# Dataset contains PS3, PS4, PS5 game sales across regions

THE SCOPE OF THIS PERSONAL PROJECT IS TO CREATE A DATA PIPELINE

LOAD -> EXPLORE -> CLEAN -> TRANSFROM -> ANALYZE -> VISUALIZE

1) LOAD THE DATA

In [440]:
import pandas as pd

In [441]:
import logging

logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
logger = logging.getLogger(__name__)

In [442]:
logger.info("STARTING THE PIPELINE")
logger.info("-"*20)

INFO:__main__:STARTING THE PIPELINE
INFO:__main__:--------------------


In [443]:

logger.info("Loading data from CSV")
input_file ="PlayStation Sales and Metadata (PS3PS4PS5) (Oct 2025).csv"
try:
    df_playstation_raw = pd.read_csv(input_file)
    logger.info("SUCCES! Loading data from CSV")
    logger.info(f"The DataFrame contains {len(df_playstation_raw)} rows and {len(df_playstation_raw.columns)} columns")
    
except FileNotFoundError:
    logger.error("The file doesn't exist")


INFO:__main__:Loading data from CSV
INFO:__main__:SUCCES! Loading data from CSV
INFO:__main__:The DataFrame contains 4963 rows and 18 columns


In [444]:
logger.info("Loading data for a comparison df")
df_playstation_raw_initial = pd.read_csv(input_file)
logger.info("Done! loading data for a comparison df")

INFO:__main__:Loading data for a comparison df
INFO:__main__:Done! loading data for a comparison df


UNDERSTANDING THE DATA

In [445]:
df_playstation_raw.describe()

Total Shipped   Total Sales      NA Sales     PAL Sales   Japan Sales  \
count   4.963000e+03  4.963000e+03  4.963000e+03  4.963000e+03  4.963000e+03   
mean    9.352005e+04  3.008906e+05  1.178199e+05  1.170482e+05  2.178723e+04   
std     8.977504e+05  1.055158e+06  4.052999e+05  4.946509e+05  8.250273e+04   
min     0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
25%     0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
50%     0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
75%     0.000000e+00  1.600000e+05  7.000000e+04  3.000000e+04  0.000000e+00   
max     2.269000e+07  2.032000e+07  6.370000e+06  9.850000e+06  2.170000e+06   

        Other Sales       rating  ratings_count   metacritic  
count  4.963000e+03  2847.000000    2847.000000  1530.000000  
mean   4.426154e+04     2.762455     269.917106    73.826144  
std    1.639967e+05     1.531923     618.110338    11.532527  
min    0.000000e+00     0.000000       0.000000    21.000000  
25%    0.000000e+00     2.400000       8.000000    68.000000  
50%    0.000000e+00     3.380000      44.000000    76.000000  
75%    2.000000e+04     3.830000     205.500000    82.000000  
max    3.120000e+06     4.760000    7225.000000    97.000000

In [446]:
df_playstation_raw.dtypes

Game              object
Console           object
Name              object
Publisher         object
Developer         object
Total Shipped    float64
Total Sales      float64
NA Sales         float64
PAL Sales        float64
Japan Sales      float64
Other Sales      float64
Release Date      object
Last Update       object
rating           float64
ratings_count    float64
metacritic       float64
genres            object
platforms         object
dtype: object

I SAW THAT RELEASE DATE WAS AN OBJECT, SO FIRST THINK FIRST I CHANGE THAT IN A DATE TIME TYPE SO WE CAN USE IT MOREA EASILY IF WE WANT TO DO SOME ANALYZES ON DATES

In [447]:

logger.info("Converting 'Release Date' column to date")
df_playstation_raw["Release Date"] = pd.to_datetime(df_playstation_raw['Release Date'], errors='coerce')
logger.info("Done! Converting 'Release Date' column to date")

INFO:__main__:Converting 'Release Date' column to date
INFO:__main__:Done! Converting 'Release Date' column to date


In [448]:
logger.info("Creating Year and Month features and transforming them into 'int'")
df_playstation_raw["Release Year"] = df_playstation_raw["Release Date"].dt.year.astype('Int64')
df_playstation_raw["Release Month"] = df_playstation_raw["Release Date"].dt.month.astype('Int64')
logger.info("Done! Creating Year and Month features and transforming them into 'int'")


INFO:__main__:Creating Year and Month features and transforming them into 'int'
INFO:__main__:Done! Creating Year and Month features and transforming them into 'int'


NOW I'VE SEPARATED THE MONTHS AND YEAR AS FEATURES. IT WILL BE FASTER TO ACCES THOSE DATA WHEN WE ARE DOING DATA EXPLORATION

FINDING DATA THAT IS NULL, 0 or NaN

In [449]:
df_playstation_raw.isnull().sum()

Game                0
Console             0
Name                0
Publisher           0
Developer           0
Total Shipped       0
Total Sales         0
NA Sales            0
PAL Sales           0
Japan Sales         0
Other Sales         0
Release Date     1437
Last Update      2162
rating           2116
ratings_count    2116
metacritic       3433
genres           2160
platforms        2116
Release Year     1437
Release Month    1437
dtype: int64

In [450]:
logger.info("Visualization of first and last game for each console")
described_data_console = df_playstation_raw.groupby('Console').describe() 
df_years_on_console = pd.DataFrame(columns=["First game", "Last game"])
df_years_on_console["First game"] = described_data_console["Release Year"]["min"]
df_years_on_console["Last game"] = described_data_console["Release Year"]["max"]
logger.info("Done! Visualization of first and last game for each console")
df_years_on_console

INFO:__main__:Visualization of first and last game for each console
INFO:__main__:Done! Visualization of first and last game for each console


First game  Last game
Console                       
PS3          2006.0     2020.0
PS4          2013.0     2023.0
PS5          2015.0     2026.0

We can see that for PS3 we had games made between 2006 and 2020, for PS4 between 2013 and 2023 and for PS5 from 2015 until today

In [451]:
df_playstation_raw.isnull().sum()

Game                0
Console             0
Name                0
Publisher           0
Developer           0
Total Shipped       0
Total Sales         0
NA Sales            0
PAL Sales           0
Japan Sales         0
Other Sales         0
Release Date     1437
Last Update      2162
rating           2116
ratings_count    2116
metacritic       3433
genres           2160
platforms        2116
Release Year     1437
Release Month    1437
dtype: int64

In [452]:
df_playstation_raw.dtypes

Game                     object
Console                  object
Name                     object
Publisher                object
Developer                object
Total Shipped           float64
Total Sales             float64
NA Sales                float64
PAL Sales               float64
Japan Sales             float64
Other Sales             float64
Release Date     datetime64[ns]
Last Update              object
rating                  float64
ratings_count           float64
metacritic              float64
genres                   object
platforms                object
Release Year              Int64
Release Month             Int64
dtype: object

Some missing values from our "float64" type features can be added from the median values, based and grouped by the type of console each missing value refers to

In [453]:
logger.info("Setting missing values for type Float64 to MEDIAN grouped by 'Console'")
numeric_with_median = ["Release Month", "Release Year","rating","ratings_count","metacritic"]
for data in numeric_with_median:
    df_playstation_raw[data] = df_playstation_raw[data].astype("float64")
    df_playstation_raw[data] = df_playstation_raw.groupby("Console")[data].transform(lambda x:x.fillna(x.median())).round()
logger.info("DONE! Setting missing values for type Float64 to MEDIAN grouped by 'Console'")
df_playstation_raw.isnull().sum()


INFO:__main__:Setting missing values for type Float64 to MEDIAN grouped by 'Console'
INFO:__main__:DONE! Setting missing values for type Float64 to MEDIAN grouped by 'Console'


Game                0
Console             0
Name                0
Publisher           0
Developer           0
Total Shipped       0
Total Sales         0
NA Sales            0
PAL Sales           0
Japan Sales         0
Other Sales         0
Release Date     1437
Last Update      2162
rating              0
ratings_count       0
metacritic          0
genres           2160
platforms        2116
Release Year        0
Release Month       0
dtype: int64

We took care of the float type data, no we are left with only genres, platform, release date and last update (the last 2 will be delete because we will have year and month features for both of them)

In [454]:
logger.info("Deteleting the 'Release Date' feature")
del df_playstation_raw["Release Date"]
logger.info("DONE! Deteleting the 'Release Date' feature")

INFO:__main__:Deteleting the 'Release Date' feature
INFO:__main__:DONE! Deteleting the 'Release Date' feature


In [455]:
df_playstation_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4963 entries, 0 to 4962
Data columns (total 19 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Game           4963 non-null   object 
 1   Console        4963 non-null   object 
 2   Name           4963 non-null   object 
 3   Publisher      4963 non-null   object 
 4   Developer      4963 non-null   object 
 5   Total Shipped  4963 non-null   float64
 6   Total Sales    4963 non-null   float64
 7   NA Sales       4963 non-null   float64
 8   PAL Sales      4963 non-null   float64
 9   Japan Sales    4963 non-null   float64
 10  Other Sales    4963 non-null   float64
 11  Last Update    2801 non-null   object 
 12  rating         4963 non-null   float64
 13  ratings_count  4963 non-null   float64
 14  metacritic     4963 non-null   float64
 15  genres         2803 non-null   object 
 16  platforms      2847 non-null   object 
 17  Release Year   4963 non-null   float64
 18  Release 

In [456]:
logger.info("Transforming 'Last Update' datatype into date + creating Year and Month features and transforming them into 'int'")
df_playstation_raw["Last Update"] = pd.to_datetime(df_playstation_raw['Last Update'], errors='coerce')
df_playstation_raw["Last Update Year"] = df_playstation_raw["Last Update"].dt.year
df_playstation_raw["Last Update Month"] = df_playstation_raw["Last Update"].dt.month
logger.info("Done! Transforming 'Last Update' datatype into date + creating Year and Month features and transforming them into 'int'")

INFO:__main__:Transforming 'Last Update' datatype into date + creating Year and Month features and transforming them into 'int'
INFO:__main__:Done! Transforming 'Last Update' datatype into date + creating Year and Month features and transforming them into 'int'


In [457]:
logger.info("Filling missing with median grouped by some features")

last_update = ["Last Update Year","Last Update Month"]
def median_for_missing_numeric(df = None, group = None, columns = None):
    for column in columns:
        df[column] = df.groupby(group)[column].transform(lambda x:x.fillna(x.median()))
        df[column] = df[column].fillna(df[column].median())
    return df
logger.info("DONE! Filling missing with median grouped by some features")
logger.info("Filling missing with global median")
df_playstation_raw = median_for_missing_numeric(df=df_playstation_raw,group=["Developer", "Publisher"], columns=last_update)
logger.info("DONE! Filling missing with global median")
df_playstation_raw.info()

INFO:__main__:Filling missing with median grouped by some features
INFO:__main__:DONE! Filling missing with median grouped by some features
INFO:__main__:Filling missing with global median
INFO:__main__:DONE! Filling missing with global median


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4963 entries, 0 to 4962
Data columns (total 21 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Game               4963 non-null   object        
 1   Console            4963 non-null   object        
 2   Name               4963 non-null   object        
 3   Publisher          4963 non-null   object        
 4   Developer          4963 non-null   object        
 5   Total Shipped      4963 non-null   float64       
 6   Total Sales        4963 non-null   float64       
 7   NA Sales           4963 non-null   float64       
 8   PAL Sales          4963 non-null   float64       
 9   Japan Sales        4963 non-null   float64       
 10  Other Sales        4963 non-null   float64       
 11  Last Update        2801 non-null   datetime64[ns]
 12  rating             4963 non-null   float64       
 13  ratings_count      4963 non-null   float64       
 14  metacrit

The missing values for 'LAST UPDATE' YEAR and 'LAST UPDATE MONTH' are completed with the median grouped by developer and publisher. If there are still some few missing values they will be assigned to the global median value. 

We have a total of 4963 rows, almost half of the total values for platforms and genres are missing so we cannot dorm those row because we will lose a good portion of the data

Creating a functin that will asign missing data for object type data to the mode found when we group data based on Publisher and Developer.

In [458]:
def set_missing_values_to_mode (df=None, column = None, group=None, fallback_global = True):
    logger.info("'set_missing_values_to_mode' function called")
    result = (df.groupby(group)[column].transform(lambda x: x.fillna(x.mode()[0]) if not x.mode().empty else x)) 
    # THIS VERIFIES IF THERE IS A          
    # GLOBAL MODE, IF IT IS WE USE THAT AS A DATA FOR OUR PLATFORM FEATURE
    if not df[column].mode().empty:
        global_item = df[column].mode()[0]
        result = df[column].fillna(global_item)
    logger.info("We have a result for'set_missing_values_to_mode' function called")
    return result

This function can pe called for the platforms and genres feature

In [459]:
logger.info("Filling missing object type data with the mode value")
df_playstation_raw["platforms"] = set_missing_values_to_mode(df=df_playstation_raw, column="platforms", group=["Publisher", "Developer"])
df_playstation_raw["genres"] = set_missing_values_to_mode(df=df_playstation_raw, column="genres", group=["Publisher", "Developer"])
logger.info("DONE! Filling missing object type data with the mode value")
df_playstation_raw.info()

INFO:__main__:Filling missing object type data with the mode value
INFO:__main__:'set_missing_values_to_mode' function called
INFO:__main__:We have a result for'set_missing_values_to_mode' function called
INFO:__main__:'set_missing_values_to_mode' function called
INFO:__main__:We have a result for'set_missing_values_to_mode' function called
INFO:__main__:DONE! Filling missing object type data with the mode value


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4963 entries, 0 to 4962
Data columns (total 21 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Game               4963 non-null   object        
 1   Console            4963 non-null   object        
 2   Name               4963 non-null   object        
 3   Publisher          4963 non-null   object        
 4   Developer          4963 non-null   object        
 5   Total Shipped      4963 non-null   float64       
 6   Total Sales        4963 non-null   float64       
 7   NA Sales           4963 non-null   float64       
 8   PAL Sales          4963 non-null   float64       
 9   Japan Sales        4963 non-null   float64       
 10  Other Sales        4963 non-null   float64       
 11  Last Update        2801 non-null   datetime64[ns]
 12  rating             4963 non-null   float64       
 13  ratings_count      4963 non-null   float64       
 14  metacrit

In [460]:
logger.info("deleting the 'Last Update' feature")
del df_playstation_raw["Last Update"]
logger.info("DONE! deleting the 'Last Update' feature")

INFO:__main__:deleting the 'Last Update' feature
INFO:__main__:DONE! deleting the 'Last Update' feature


In [461]:
df_playstation_clean = df_playstation_raw.copy()

IF WE HAVE THE CLEAN DATA FATAFRAME, LET'S CONTINUE USING THAT NAME FOR A BETTER UNDERSTANDING, AND SAVE IT IN A CSV FILE BECAUSE NOW WE HAVE CLEAN DATA

In [462]:
logger.info("Save the outputdata in a CSV file")
output_clean_data = df_playstation_clean.copy()
output_clean_data.to_csv("clean_data/clean_PlayStation_Data_report.csv")
logger.info("SUCCESS! Save the outputdata in a CSV file")

INFO:__main__:Save the outputdata in a CSV file
INFO:__main__:SUCCES! Save the outputdata in a CSV file


Creating a Data Quality Report that shows how the raw data looks compared with the clean data that we obtained

In [468]:
def data_quality_report(df, name="Dataset"):
    logger.info("Starting the call for the data_quality_report")
    print(f"\n{'='*50}")
    print(f"DATA QUALITY REPORT: {name}")
    print(f"{'='*50}\n")
    
    print(f"Total rows: {len(df)}")
    print(f"Total columns: {len(df.columns)}")
    
    print(f"\n{'─'*50}")
    print("COMPLETENESS")
    print(f"{'─'*50}")
    missing = df.isnull().sum()
    missing_pct = (missing / len(df) * 100).round(2)
    
    quality_df = pd.DataFrame({
        'Missing': missing,
        'Missing %': missing_pct,
        'Complete %': 100 - missing_pct
    })
    quality_df = quality_df[quality_df['Missing'] > 0].sort_values('Missing', ascending=False)
    
    if len(quality_df) > 0:
        print(quality_df)
    else:
        print("✓ No missing values!")
    
    print(f"\n{'─'*50}")
    print("DUPLICATES")
    print(f"{'─'*50}")
    duplicates = df.duplicated().sum()
    print(f"Duplicate rows: {duplicates} ({duplicates/len(df)*100:.2f}%)")
    logger.info("SUCCESS! Starting the call for the data_quality_report")
    return quality_df

In [469]:
logger.info("Using a call for data_quality_report function")
data_quality_report(df_playstation_raw_initial, "PlayStation Raw Data")
logger.info("DONE! Using a call for data_quality_report function")

INFO:__main__:Using a call for data_quality_report function
INFO:__main__:Starting the call for the data_quality_report
INFO:__main__:SUCCESS! Starting the call for the data_quality_report
INFO:__main__:DONE! Using a call for data_quality_report function



DATA QUALITY REPORT: PlayStation Raw Data

Total rows: 4963
Total columns: 18

──────────────────────────────────────────────────
COMPLETENESS
──────────────────────────────────────────────────
               Missing  Missing %  Complete %
metacritic        3433      69.17       30.83
Last Update       2162      43.56       56.44
genres            2160      43.52       56.48
rating            2116      42.64       57.36
ratings_count     2116      42.64       57.36
platforms         2116      42.64       57.36
Release Date      1437      28.95       71.05

──────────────────────────────────────────────────
DUPLICATES
──────────────────────────────────────────────────
Duplicate rows: 0 (0.00%)


In [470]:
logger.info("Using a call for data_quality_report function")
data_quality_report(df_playstation_clean, "PlayStation Clean Data")
logger.info("DONE! Using a call for data_quality_report function")

INFO:__main__:Using a call for data_quality_report function
INFO:__main__:Starting the call for the data_quality_report
INFO:__main__:SUCCESS! Starting the call for the data_quality_report
INFO:__main__:DONE! Using a call for data_quality_report function



DATA QUALITY REPORT: PlayStation Clean Data

Total rows: 4963
Total columns: 20

──────────────────────────────────────────────────
COMPLETENESS
──────────────────────────────────────────────────
✓ No missing values!

──────────────────────────────────────────────────
DUPLICATES
──────────────────────────────────────────────────
Duplicate rows: 0 (0.00%)


We managed to keep all the initial rows and just find a solution for the missing values inside those rows

## COMPLETE THE FIRST PART OF THE PIPELINE:
##  LOAD -> EXPLLORE -> CLEAN -> TRANSFORM

# ANALYZE 

In [471]:
import matplotlib as mpl
import matplotlib.pyplot as plt
import numpy as np
import sqlite3

In [483]:
logger.info("creating a db and a connection with sqlite for SQL interogations")
conn = sqlite3.connect('playstation_sales.db')
logger.info("DONE! creating a db and a connection with sqlite for SQL interogations")

INFO:__main__:creating a db and a connection with sqlite for SQL interogations
INFO:__main__:DONE! creating a db and a connection with sqlite for SQL interogations


In [484]:
logger.info("creating the table and connecting it to the database")
table_name = "PlayStation_Games"
df_playstation_clean.to_sql(
    name=table_name,
    con=conn,
    if_exists='replace',
    index=False
)
logger.info("DONE! creating the table and connecting it to the database")

INFO:__main__:creating the table and connecting it to the database
INFO:__main__:DONE! creating the table and connecting it to the database


In [486]:
logger.info("Checking if the connection is working properly")
query_verify = "SELECT COUNT(*) as Count FROM PlayStation_Games"
row_count = pd.read_sql(query_verify, conn)
logger.info("DONE! Checking if the connection is working properly")
row_count

INFO:__main__:Checking if the connection is working properly
INFO:__main__:DONE! Checking if the connection is working properly


Count
0   4963

1) Which console has the most games?

In [494]:
query_most_games = """
    WITH CTE_most_games as(SELECT Console, COUNT(*) as Nr_of_games
    FROM PlayStation_Games
    GROUP By Console)

    SELECT Console, MAX(Nr_of_games) as Most_Games
    FROM CTE_most_games
    
"""
logger.info("sql interogation on the table from database")
nr_of_games_per_console = pd.read_sql(query_most_games,conn)
nr_of_games_per_console.set_index("Console",inplace=True)
logger.info("DONE! sql interogation on the table from database")
nr_of_games_per_console

INFO:__main__:sql interogation on the table from database
INFO:__main__:DONE! sql interogation on the table from database


Most_Games
Console            
PS4            1991

2) Top 10 games with the most copies sold

In [ ]:
query_games_max_sold = """
SELECT Game, [Total Sales]
FROM PlayStation_Games
ORDER BY [Total Sales] DESC
LIMIT 10
"""
logger.info("sql interogation on the table from database")
max_games_sold = pd.read_sql(query_games_max_sold,conn)
logger.info("Done! sql interogation on the table from database")
max_games_sold

Game  Total Sales
0              PS3_Grand Theft Auto V   20320000.0
1              PS4_Grand Theft Auto V   19390000.0
2       PS4_Call of Duty: Black Ops 3   15090000.0
3           PS4_Red Dead Redemption 2   13940000.0
4      PS3_Call of Duty: Black Ops II   13800000.0
5              PS4_Call of Duty: WWII   13400000.0
6  PS3_Call of Duty: Modern Warfare 3   13350000.0
7         PS3_Call of Duty: Black Ops   12670000.0
8                         PS4_FIFA 18   11800000.0
9                         PS4_FIFA 17   10940000.0

3) What are the top 5 best games for each console?

In [ ]:
query_games_for_console = """
SELECT Console, Game, [Total Sales]
FROM(
    SELECT Console, Game, "Total Sales",
        ROW_NUMBER() OVER (PARTITION BY Console ORDER BY "Total Sales" DESC) as rank
    FROM PlayStation_Games)
WHERE rank <=5
ORDER BY Console, rank
"""
logger.info("sql interogation on the table from database")
best_games_for_console = pd.read_sql(query_games_for_console,conn)
logger.info("DONE! sql interogation on the table from database")
best_games_for_console

Console                                 Game  Total Sales
0      PS3               PS3_Grand Theft Auto V   20320000.0
1      PS3       PS3_Call of Duty: Black Ops II   13800000.0
2      PS3   PS3_Call of Duty: Modern Warfare 3   13350000.0
3      PS3          PS3_Call of Duty: Black Ops   12670000.0
4      PS3   PS3_Call of Duty: Modern Warfare 2   10610000.0
5      PS4               PS4_Grand Theft Auto V   19390000.0
6      PS4        PS4_Call of Duty: Black Ops 3   15090000.0
7      PS4            PS4_Red Dead Redemption 2   13940000.0
8      PS4               PS4_Call of Duty: WWII   13400000.0
9      PS4                          PS4_FIFA 18   11800000.0
10     PS5  PS5_Shadow of Conspiracy: Section 2          0.0
11     PS5                        PS5_SacriFire          0.0
12     PS5                     PS5_Purpose 1951          0.0
13     PS5   PS5_Warhammer 40,000: Rogue Trader          0.0
14     PS5       PS5_Punch Club 2: Fast Forward          0.0

4) What are the top 15 Publishers based on total sales?

In [ ]:
query_15_publishers = """
    SELECT Publisher, SUM([Total Sales]) as All_Sales
    FROM PlayStation_Games
    GROUP BY Publisher
    ORDER BY All_Sales DESC
    LIMIT 15
"""
logger.info("sql interogation on the table from database")
total_biggest_15_publishers = pd.read_sql(query_15_publishers,conn)
logger.info("Done! sql interogation on the table from database")
total_biggest_15_publishers

Publisher    All_Sales
0                               Activision  205940000.0
1                          Electronic Arts  153760000.0
2                                EA Sports  129990000.0
3                                  Ubisoft  128370000.0
4              Sony Computer Entertainment  103700000.0
5                           Rockstar Games   82280000.0
6                              Square Enix   58150000.0
7           Sony Interactive Entertainment   57870000.0
8                       Bethesda Softworks   53740000.0
9   Warner Bros. Interactive Entertainment   51850000.0
10                                  Capcom   43220000.0
11                               2K Sports   40630000.0
12                                    Sega   31080000.0
13                      Namco Bandai Games   29530000.0
14                                     THQ   27160000.0

5) What are the top 10 Developers based on average rating

In [ ]:
query_dev_top_10_devs = """
    SELECT Developer, AVG(rating) as average_rating
    FROM PlayStation_Games
    GROUP BY Developer
    ORDER BY average_rating DESC
    LIMIT 10    
"""
logger.info("sql interogation on the table from database")
top_10_devs_rating = pd.read_sql(query_dev_top_10_devs,conn)
logger.info("Done! sql interogation on the table from database")
top_10_devs_rating

Developer  average_rating
0            Valve Software        5.000000
1      Sandfall Interactive        5.000000
2   SIE Santa Monica Studio        5.000000
3           Edmund McMillen        5.000000
4                     EA UK        5.000000
5     CD Projekt Red Studio        4.500000
6  Sucker Punch Productions        4.166667
7           à la mode games        4.000000
8           thatgamecompany        4.000000
9                  tha ltd.        4.000000

6) What developers has the most games made

In [ ]:
query_dev_most_games = """
    SELECT Developer, COUNT(Game) as Nr_of_games  
    FROM PlayStation_Games
    GROUP BY DEVELOPER
    HAVING Developer != "Unknown"
    ORDER BY Nr_of_games DESC
    LIMIT 3  
"""
logger.info("sql interogation on the table from database")
dev_with_most_games = pd.read_sql(query_dev_most_games,conn)
logger.info("Done! sql interogation on the table from database")
dev_with_most_games

Developer  Nr_of_games
0       Capcom           82
1  Square Enix           67
2  Omega Force           55

In [517]:
df_playstation_clean.head(1)

Game Console                Name       Publisher  \
0  PS3_Grand Theft Auto V     PS3  Grand Theft Auto V  Rockstar Games   

        Developer  Total Shipped  Total Sales   NA Sales  PAL Sales  \
0  Rockstar North            0.0   20320000.0  6370000.0  9850000.0   

   Japan Sales  Other Sales  rating  ratings_count  metacritic  genres  \
0     990000.0    3120000.0     4.0         7225.0        92.0  Action   

                                           platforms  Release Year  \
0  PC, PlayStation 5, Xbox One, PlayStation 4, Xb...        2013.0   

   Release Month  Last Update Year  Last Update Month  
0            9.0            2018.0                1.0